packages/grafana-ui/src/components/Forms/Form.mdx

```diff
@@ -182,6 +185,92 @@ const defaultValues: FormDto {
 </Forms.Form>
 ```
 
+### Validation
+
+Validation can be performed either synchronously or asynchronously. What's important here is that the validation function must return either a `boolean` or a `string`.
+
+#### Basic required example
+
+```jsx
+<Forms.Form ...>{
+  ({register, errors}) => (
+    <>
+      <Forms.Field invalid={!!errors.name} error={errors.name && 'Name is required'}
+      <Forms.Input
+        defaultValue={default.name}
+        name="name"
+        ref={register({ required: true })}
+      />
+    </>
+  )}
+</Forms.Form>
+```
+
+#### Required with synchronous custom validation
+
+One important thing to note is that if you want to provide different error messages for different kind of validation errors you'll need to return a `string` instead of a `boolean`.
+
+```jsx
+<Forms.Form ...>{
+  ({register, errors}) => (
+    <>
+      <Forms.Field invalid={!!errors.name} error={errors.name && 'Name is required'}
```

Comment:
Will this return the right message in case of custom validation fails? 
I think it'd be `error={errors.name?.message}` and then in required it would return the message like `required: 'Name is required',`

Reply:
ops